src/fetchBaseQuery.ts

```diff
@@ -1,28 +1,73 @@
 import { QueryApi } from './buildThunks';
+import { isPlainObject, joinUrls } from './utils';
 
 interface FetchArgs extends RequestInit {
   url: string;
+  params?: URLSearchParams | Record<string, string>;
+  body?: any;
+  responseHandler?: 'json' | 'text' | ((response: Response) => Promise<any>);
+  validateStatus?: (response: Response, body: any) => boolean;
 }
 
-export function fetchBaseQuery({ baseUrl }: { baseUrl: string } = { baseUrl: '' }) {
+const defaultValidateStatus = (response: Response) => response.status >= 200 && response.status <= 299;
```

Comment:
I just jotted this down quickly, but it would be worth to check what fetch does in a 3xx case. Follow the redirect? If so, return the redirect target's return code?

Reply:
Fetch follows redirect by default 🎉